packages/contracts/contracts/minter-suite/Minters/MinterSetPriceERC20V5.sol

```diff
@@ -477,6 +482,22 @@ contract MinterSetPriceERC20V5 is ReentrancyGuard, ISharedMinterV0 {
             coreContract: coreContract
         });
 
+        // get the currency address configured on the project
+        (address configuredCurrencyAddress, ) = SplitFundsLib
+            .getCurrencyInfoERC20(projectId, coreContract);
+
+        // validate that the currency address matches the project configured currency
+        require(
+            currencyAddress == configuredCurrencyAddress,
+            "Currency addresses must match"
+        );
+
+        // validate that the specified maximum price is greater than or equal to the price per token
+        require(
+            maxPricePerToken >= pricePerTokenInWei,
+            "Only max price gte token price"
+        );
+
         // @dev revert occurs during payment split if ERC20 token is not
         // configured (i.e. address(0)), so check is not performed here
```

Comment:
nit - might make sense to move this comment immediately after the currency check comment

Reply:
updated in [41e4959](https://github.com/ArtBlocks/artblocks-contracts/pull/1141/commits/41e4959305fcb226e09b0aae86cc6c0b5860582f)